coding-articles/Article.md

```diff
@@ -0,0 +1,584 @@
+````
+
+Inside the file, put the JSON RPC URL for your Blockchain network
+```
+{
+"jsonRpcUrl": "https://polygon-mumbai.g.alchemy.com/v2/<Your key>"
+}
+```
+Now you have your bot listening to the Mumbai network to test run
+```
+npm start
+```
+Should give output like this
+
+![project-root-files](./images/bot-output.png)
+<div style="text-align:center">If you searched for any tx hash, you should find it on https://mumbai.polygonscan.com</div>
+<br/>
+
+As you see above, no findings will be emitted because the contract address is still the wrong one from the Ethereum Mainnet network. Let's add our own smart contract in Network Manager.
+
+## Network Manager: Specify the smart contract addresses for different networks we want to monitor
+
+The network manager has two main purposes. The first one is to map data to its equivalent blockchain network. So if I have a contract address `0xa…` deployed on Ethereum and a contract address `0x0b…` deployed on Polygon. You can get each respective address without changing the code. It maps the addresses and any other equivalent data to  chainId this map name is `networkData`
+
+Create a file for the networkData Map
+
+```
+nano src/network.ts
+```
+
+Inside `network.ts` add
+
+```
+export interface NetworkData {
+  address: string; // The smart contract address
+  minimumDepositAmount: number; 
+  blacklistedAddresses: string[];
+}
+export const networkData: Record<number, NetworkData> = {
+  80001: {
+    address: "0xa87db9fe057cff6e296586bec6a6982a5a9b44b0",
+    minimumDepositAmount: 0.01,
+    blacklistedAddresses: [
+      "0x7c71a3d85a8d620eeab9339cce776ddc14a8129c",
+      "0x17156c0cf9701b09114cb3619d9f3fd937caa3a8",
+    ],
+  },
+/**
+    Bonus: Add Görli network object here, note that for  Ethereum 0.01 ETH maybe a good amount so the minimumDepositAmount should be less than 0.01 ETH
+*/
+};
+```
+The second purpose is to use the network manager object to initialize the provider object for the network connection. (If you don't know what a network provider check [this](https://docs.ethers.io/v5/api/providers/#providers)).
+
+To do so, we need to install [forta-agent-tools](https://www.npmjs.com/package/forta-agent-tools)
+
+```
+npm i forta-agent-tools
+```
+
+Go to `src/agent.ts` 
+
+**Delete all** content inside it and import the `NetworkManager` class from the installed package
+
+```
+import { NetworkManager } from "forta-agent-tools";
+```
+
+Create the object with the `networkData`. I created on the [network.ts](./code/src/network.ts)
+
+```
+// below the import NetworkManager
+import { networkData } from "./network";
+const networkManager = new NetworkManager(networkData); //global var
+```
+
+To add the provider, we need to call `networkManager.init(<provider>)` , but we need this to be called only one time when the bot is **initialized**. Forta has a built-in function to [initialize](https://docs.forta.network/en/latest/sdk/#handlers)
+
+```
+import ethers from "forta-agent"
```

Comment:
ethers has been imported from "forta agent" here, but it is not present in the summary of the agent.ts file at line 212.

Reply:
That's true. fixed. Thank you!